databricks/koalas/frame.py

```diff
@@ -15,55 +15,62 @@
 #
 
 """
-A base class to be monkey-patched to DataFrame to behave similar to pandas DataFrame.
+A wrapper class for Spark DataFrame to behave similar to pandas DataFrame.
 """
+from decorator import dispatch_on
 from functools import reduce
 
 import numpy as np
 import pandas as pd
-from pyspark.sql import Column, DataFrame, functions as F
+from pyspark import sql as spark
+from pyspark.sql import functions as F
 from pyspark.sql.types import StructType, to_arrow_type
 from pyspark.sql.utils import AnalysisException
 
 from databricks.koalas.dask.compatibility import string_types
 from databricks.koalas.dask.utils import derived_from
-from databricks.koalas.generic import _Frame, anchor_wrap, max_display_count
+from databricks.koalas.generic import _Frame, max_display_count
 from databricks.koalas.metadata import Metadata
 from databricks.koalas.missing.frame import _MissingPandasLikeDataFrame
 from databricks.koalas.selection import SparkDataFrameLocator
 
 
-class PandasLikeDataFrame(_Frame, _MissingPandasLikeDataFrame):
-    """
-    Methods that are relevant to dataframes.
-    """
+def default_session():
+    return spark.SparkSession.builder.getOrCreate()
 
-    def __init__(self):
-        """ Define additional private fields.
-
-        * ``_pandas_metadata``: The metadata which stores column fields, and index fields and names.
-        """
-        self._pandas_metadata = None
 
-    @property
-    def _metadata(self):
-        if not hasattr(self, '_pandas_metadata') or self._pandas_metadata is None:
-            self._pandas_metadata = Metadata(column_fields=self.schema.fieldNames())
-        return self._pandas_metadata
+class DataFrame(_Frame, _MissingPandasLikeDataFrame):
```

Comment:
Please document all the hidden fields that you have, it is not easy to track with the dispatch (`_sdf`, `_metadata`). This is important for the reviewers.